File: Exams/E3-RestAPI/JPA/src/main/java/com/danielmunoz/psp/jpa/data/Song.java
package com.danielmunoz.psp.jpa.data;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Song {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String name;

    @Column
    private LocalTime time;

    @ManyToOne
    @JoinColumn(name = "album_id")
    @JsonIgnore
    private Album album;

}

File: Exams/E3-RestAPI/src/main/kotlin/com/raks/jpa/entity/Song.kt
package com.raks.jpa.entity

import jakarta.persistence.*
import java.time.LocalTime

@Entity
@Table(name = "songs")
open class Song {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    open var id:    Int?       = null

    @Column(name = "name", nullable = false)
    open var name:  String?    = null

    @Column(name = "time", nullable = false)
    open var time:  LocalTime? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "album_id", nullable = false)
    open var album: Album?     = null

}